plugin.js

```diff
@@ -90,13 +89,21 @@ function compare (a, b) {
   }
 }
 
-function plugin (fastify, options, next) {
-  fastify.addHook('onRequest', factory(options))
-  next()
+function plugin (fastify, options, done) {
+  options = Object.assign({ addHook: true }, options)
+
+  if (options.addHook === true) {
+    fastify.addHook('onRequest', verifyBearerAuthFactory(options))
+  } else {
+    fastify.decorate('verifyBearerAuthFactory', verifyBearerAuthFactory)
```

Comment:
what is the value in exposing this as a decorator instead of exporting the function directly?

Reply:
You can access it through `fastify.verifyBearerAuthFactory` and `fastify.verifyBearerAuth`. or even `this.verifyBearerAuthFactory` and `this.verifyBearerAuth`.

Reply:
FYI, you can access the `factory` function without this PR.
```JS
import { internals } from 'fastify-bearer-auth`
const verifyBearerAuthFactory = internals.factory
```